examples/game_data/main.rs

```diff
@@ -0,0 +1,583 @@
+            if time.frame_number() % 20 == 0 {
+                let fps = fps_counter.sampled_fps();
+                fps_display.text = format!("FPS: {:.*}", 2, fps);
+            }
+        }
+    }
+}
+
+#[derive(Default)]
+struct Loading {
+    progress: Option<ProgressCounter>,
+}
+struct Main;
+struct Paused;
+
+#[derive(Default)]
+struct Tag;
+
+impl Component for Tag {
+    type Storage = NullStorage<Self>;
+}
+
+impl<'a, 'b> State<GameData<'a, 'b>> for Loading {
+    fn on_start(&mut self, data: StateData<GameData>) {
+        data.world.register::<Tag>();
+        let (progress, assets) = load_assets(&data.world);
+        self.progress = Some(progress);
+        let fps_display = data.world
+            .create_entity()
+            .with(UiTransform::new(
+                "fps".to_string(),
+                100.,
+                25.,
+                1.,
+                200.,
+                50.,
+                0,
+            ))
+            .with(UiText::new(
+                assets.font.clone(),
+                "N/A".to_string(),
+                [1.0, 1.0, 1.0, 1.0],
+                25.,
+            ))
+            .with(Anchored::new(Anchor::TopLeft))
+            .build();
+
+        data.world
+            .create_entity()
+            .with(UiTransform::new(
+                "fps".to_string(),
+                0.,
+                0.,
+                1.,
+                200.,
+                50.,
+                0,
+            ))
+            .with(UiText::new(
+                assets.font.clone(),
+                "Loading".to_string(),
+                [1.0, 1.0, 1.0, 1.0],
+                25.,
+            ))
+            .with(Anchored::new(Anchor::Middle))
+            .with(Tag)
+            .build();
+
+        data.world.add_resource(assets);
+        data.world.add_resource::<DemoState>(DemoState {
+            light_angle: 0.0,
+            light_color: [1.0; 4],
+            ambient_light: true,
+            point_light: true,
+            directional_light: true,
+            camera_angle: 0.0,
+            fps_display,
+        });
+    }
+
+    fn handle_event(&mut self, _: StateData<GameData>, event: Event) -> Trans<GameData<'a, 'b>> {
+        match event {
+            Event::WindowEvent { event, .. } => match event {
+                WindowEvent::KeyboardInput {
+                    input:
+                        KeyboardInput {
+                            virtual_keycode: Some(VirtualKeyCode::Escape),
+                            ..
+                        },
+                    ..
+                } => Trans::Quit,
+                _ => Trans::None,
+            },
+            _ => Trans::None,
+        }
+    }
+
+    fn update(&mut self, data: StateData<GameData>) -> Trans<GameData<'a, 'b>> {
+        data.data.update(&data.world, true);
+        match self.progress.as_ref().unwrap().complete() {
+            Completion::Failed => {
+                println!("Failed loading assets");
```

Comment:
printerr! or error!

Reply:
printerr! then?

Reply:
printerr? Not a part of core rust from what I know ?

Reply:
Ya sorry its "eprintln!" actually :P